### PR TITLE
dev_doeclim: a bug fix, const parameters, updated references

### PR DESCRIPTION
**Pull request by bvegawe**
Bug fix in total heat flux calculation. Improved DOECLIM references. Made some parameters const and moved their assignment to the header.

### CODE REVIEW

File: project_files/xcode/hector.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -7,9 +7,6 @@
 	objects = {
```
Comment (cahartin) [inline project_files/xcode/hector.xcodeproj/project.pbxproj]: @bpbond and @rplzzz do we want to commit the xcode project file?

Comment (rplzzz) [reply project_files/xcode/hector.xcodeproj/project.pbxproj]: You probably should.  Windows users will need it in order to build the code, so it's best if it comes in the package.

**Comment by bvegawe** [comment]
@cahartin Hmm, actually I was previously testing this on my mac using xcode. When I now try to build on linux (using `make hector`), I get the following for all of these header initializations: `error: ISO C++ forbids initialization of member`...trying to figure out a solution where we keep them as `const`.

**Comment by cahartin** [comment]
@bvegawe let me tag @pralitp and @rplzzz to see if they know what the issue is.

**Comment by cahartin** [comment]
@bvegawe did you try make clean? and then what compiler are you using? did that change?

**Comment by bvegawe** [comment]
I guess the error depends on the version of gcc. My local cluster default is 4.4.7. When I switch to 6.1.0, the error goes away. 

Do you think that version sensitivity is a problem?

**Comment by cahartin** [comment]
Yea, we make sure to always load 6.1.0 when running GCAM/Hector.

**Comment by bvegawe** [comment]
Ah ok; I'll do the same from now on.

**Pull request merged**